main.py

```diff
@@ -185,19 +193,25 @@ def process_images():
           latest = q.get_nowait()
         except Queue.Empty:
           break
-      recognizer.detect_and_show(latest)
+      try:
+        recognizer.detect_and_show(latest)
+      except cv2.error:
+        # Sometimes, cvtcolor fails.
```

Comment:
I added this after adding the explicit check -- it happened after everything had been running for a couple of minutes. Seemed like a good thing to do to boost reliability.